### ISSUE TITLE
[network] Replace Noise with TLS

### ISSUE THREAD

**Issue by gregnazario** [aptos-labs/aptos-core#301]
To simplify key management and use generic tooling, we should move to using TLS rather than Noise for node communication.

- [ ] Investigate and choose which TLS to use e.g. Rustls
- [ ] Allow for switching of protocols from Noise to TLS (run side by side? only one?)
- [ ] Test in a local testnet to ensure that they can talk to each other
- [ ] Autogenerate static identities via TLS

Open questions:
* How do we get keys managed and stored without having to deal with entire certificates?
* Can we hand just keys through address rather than whole certificate?

**Comment by JoshLind** [comment]
For context, do we have any idea why Noise was originally chosen over TLS? Hopefully @davidiw has context 😄

**Comment by gregnazario** [comment]
tl;dr Certs are large

**Comment by davidiw** [comment]
* TLS is complex, concerns around security (downgrade attacks, etc)
* Concern around TLS being more Web PKI oriented
* WhatsApp uses it
* Attacks on certificates / parsing etc

You actually don't need to exchange certs out of band, (identity, public key) exchanges off-band are sufficient
